espnet/nets/pytorch_backend/rnn/encoders.py

```diff
@@ -32,21 +32,22 @@ def __init__(self, idim, elayers, cdim, hdim, subsample, dropout, typ="blstm"):
                 inputdim = idim
             else:
                 inputdim = hdim
-            rnn = torch.nn.LSTM(inputdim, cdim, dropout=dropout, num_layers=1, bidirectional=bidir,
-                                batch_first=True) if "lstm" in typ \
-                else torch.nn.GRU(inputdim, cdim, dropout=dropout, num_layers=1, bidirectional=bidir, batch_first=True)
+
+            RNN = torch.nn.LSTM if "lstm" in typ else torch.nn.GRU
+            rnn = RNN(inputdim, cdim, num_layers=1, bidirectional=bidir, batch_first=True)
+            num_directions = 1 + bidir
```

Comment:
This is right but the https://github.com/espnet/espnet/pull/1784/files#diff-12afbdd087fdf0d238a6a6abe4b582e0L40-L43 is more readable.
Maybe we can keep this part as it is?

Reply:
I restored the lines 40-43 to their original form